packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { useMergedRefs } from '@fluentui/react-utilities';
+import { useKeyboardNavigationState } from '@fluentui/react-focus-management';
+import { MenuTriggerState } from './MenuTrigger.types';
+import { useMenuContext } from '../../menuContext';
+
+// Helper type to select on parts of state the hook uses
+type UseTriggerElementState = Pick<MenuTriggerState, 'children'>;
+
+export const useTriggerElement = (state: UseTriggerElementState): MenuTriggerState => {
+  const triggerRef = useMenuContext(context => context.triggerRef);
+  const setOpen = useMenuContext(context => context.setOpen);
+  const open = useMenuContext(context => context.open);
+  const triggerId = useMenuContext(context => context.triggerId);
+  const on = useMenuContext(context => context.on);
+  const { isNavigatingWithKeyboard } = useKeyboardNavigationState();
+
+  const child = React.Children.only(state.children);
+
+  const triggerProps: Partial<React.HTMLAttributes<HTMLElement>> = {
+    'aria-haspopup': true,
+    'aria-expanded': open,
+    id: triggerId,
```

Comment:
This case creates an interesting problem as it overwrites user's input:

```tsx
<MenuTrigger>
  {/* ⚠ "id" from `triggerProps` will overwrite user's */}
  <button id="my-awesome-id" />
</MenuTrigger>
```

Ideally, we should reuse user's input if it's possible. Can we try to do something like this:

```tsx
const id = _.isNil(child.props.id) ? triggerId : child.props.id;
//
state.someSlot["aria-labelledby"] = id; // it's important to propagate a proper "id" back
```

Reply:
Yes, that's me just forgetting that it can be a prop too

Reply:
Tracking this as a checkbox item in #16465 I don't see a problem with doing this, but would rather  a PR to do this for both `MenuGroup` and this component together

Reply:
👍 _(I updated an item to include more details an reference this comment)_